ingredients/LSST/DC2/DRP.yaml

```diff
@@ -0,0 +1,221 @@
+description: DRP specialized for ImSim-DC2 data
```

Comment:
I am confused about the directory naming. In `ap_pipe` we have `DarkEnergyCamera`, `HyperSuprimeCam`, and `LsstCamImSim` as the three `pipelines` subdirectories. Shouldn't it be the same here in both `ingredients` and `pipelines`? I am also confused why "DC2" gets its own `DRP.yaml` in ingredients, but HSC and DECam do not.

Reply:
Second question first, because I think it's mostly covered by the other thread: DC2 got its own `DRP.yaml` ingredients because it was easy, HSC didn't because it's hard and I wanted to punt any nontrivial refactoring to another ticket, and DECam didn't because I didn't notice how easy it would be, and so I'll go fix that now.

As for the names, the ones I've used here are the "short" names used in butler data IDs and collection names.  I do think those names are preferably almost everywhere the Python class names (which I gather is what ap_pipe is currently using).  The Python class names are constrained both by our coding conventions (camelcase! no acronyms!) and greater potential for name conflicts in a way the short names generally are not, and I think the short names are much more user-friendly.  So I'm hoping we could actually change `ap_pipe` on this one.

As an aside, the LSST/DC2 name is a little weird because - being simulations - the instrument isn't really a solid identifier for the data, because ImSim can fundamentally change between versions in a way physical instruments can't.  That's why `/repo/dc2` is a different repo entirely at NCSA, and "2.2i" (the simulation run version) is used in collection names there instead of just the instrument name.

Reply:
I forget exactly why we chose the Python class names for `ap_pipe`, but I think at least part of the reason was because I kept having to look up what `instrument` the butler was governor-haranguing me to specify, and it seemed the most canonical choice. Anyway, I'm fine switching to shorter names for `HSC` and `DECam` in `ap_pipe`.

Regarding `DC2`. I enthusiastically approve of keeping fake data wholly separate from real data (separate repos, as you say). But with this pipeline and ingredient directory structure, in the future we'll have `LSST/LsstCam` alongside `LSST/DC2`? It seems wrong to have both fall under `LSST` when one is fake and one is not, and when they would never ever share a butler repo.

Reply:
@TallJimbo, is the fact that these should be short names written down anywhere other than RFC-775? I believe that standardization is much more important here than what the standard actually is.

As for DC2 pipelines, I kind of like the current arrangement, since I think the only reason we care about simulated data is as a proxy for real data, so the processing should be as similar as possible. Even if the ImSim "instrument" needs to be versioned, it's still an LSST instrument.

Reply:
> is the fact that these should be short names written down anywhere other than RFC-775?

DMTN-167 (organization of and naming conventions for data repositories) does talk about using the short names there, but I think RFC-775 is the only document that talks about this particular case at all.  Documenting this convention somewhere seems like a good idea, though; best idea I have would be README files in the `pipelines` directories of `drp_pipe` and `ap_pipe`, but I'm certainly open to other ideas, too.

> But with this pipeline and ingredient directory structure, in the future we'll have LSST/LsstCam alongside LSST/DC2

Like @kfindeisen , I do think it's appropriate to include LSST in these names, but I think @mrawls makes a good point that `LSST/LsstCam` (or `LSST/LSSTCam` with the actual short name) is a bit awkward, and just `LSST` is a little awkward, too, since it makes the real-LSST pipelines siblings of the DC2 directory.  Maybe using "Rubin" is an easy way out, even if it _isn't_ part of the short name (maybe it should be, or should have been, if timing had been different):
```
DECam/*
HSC/*
Rubin/DC2/*   # DC2 pipelines
Rubin/LSST/*   # real LSST pipelines
```
Another option would be to use LSST in the prefix for the DC2 directory, without using a level of directory structure:
```
DECam/*
HSC/*
LSST-DC2/*   # DC2 pipelines
LSST/*   # real LSST pipelines
```
Any preference or other ideas?

Reply:
I'd like to avoid special-casing (it's hard for human pipeline users to figure out, and extra-hard for machines), so I don't really like either of these. I'd prefer:

```
    DECam/*
    HSC/*
    ImSim/*
    LSSTCam/*
```

That's assuming I got the short names right, of course...

Reply:
Done; the short name for ImSim is actually `LSSTCam-imSim`, and that's what I've used.

Reply:
This is good by me. I made DM-33453 to update the ap_pipe pipelines directory structures accordingly.